### PR TITLE
[O2B-1266] Special treatment of Limited Acceptance MC Reproducible flag in QC summary

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [x] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- Added special display for Limited Acceptance MC Reproducible in QC summary
- Changed QC summary colours palette to darker one

Notable changes for developers:
- NA

Changes made to the database:
- Added monte_carlo_reproducible column to quality_control_flag_types table

### CODE REVIEW

File: lib/public/views/Runs/ActiveColumns/runDetectorsAsyncQcActiveColumns.js [hunk 1]
```diff
@@ -67,15 +72,26 @@ const runDetectorAsyncQualityDisplay = ({ dataPassId, simulationPassId }, run, d
...
+            );
         } else {
-            qcSummaryDisplay = getQcSummaryDisplayWithColor(notBadPercentageFormat, { color: QC_SUMMARY_COLORS.PARTIALLY_BAD });
+            if (badEffectiveRunCoverage === 1) {
+                qcSummaryDisplay = getQcSummaryDisplayWithColor(qcSummaryContent, { color: QC_SUMMARY_COLORS.ALL_BAD });
+            } else {
+                qcSummaryDisplay = getQcSummaryDisplayWithColor(qcSummaryContent, { color: QC_SUMMARY_COLORS.PARTIALLY_BAD });
+            }
         }
     }
```
Comment (martinboulais) [inline lib/public/views/Runs/ActiveColumns/runDetectorsAsyncQcActiveColumns.js:96]: This seriously starts to be hard to read

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/public/views/Runs/ActiveColumns/runDetectorsAsyncQcActiveColumns.js [hunk 2]
```diff
@@ -37,16 +37,21 @@ const runDetectorAsyncQualityDisplay = ({ dataPassId, simulationPassId }, run, d
...
-        h('.d-inline-block.va-t-bottom', missingVerificationsCount
-            ? tooltip(
-                h('.f7', iconWarning()),
-                `Missing ${missingVerificationsCount} verification${missingVerificationsCount > 1 ? 's' : ''}`,
-            )
-            : null),
+    const missingVerificationDisplay = h('.d-inline-block.va-t-bottom', missingVerificationsCount
+        ? tooltip(
+            h('.f7', iconWarning()),
+            `Missing ${missingVerificationsCount} verification${missingVerificationsCount > 1 ? 's' : ''}`,
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/views/Runs/ActiveColumns/runDetectorsAsyncQcActiveColumns.js:46]: applied

File: lib/public/views/Runs/ActiveColumns/runDetectorsAsyncQcActiveColumns.js [hunk 3]
```diff
@@ -67,15 +72,26 @@ const runDetectorAsyncQualityDisplay = ({ dataPassId, simulationPassId }, run, d
     let qcSummaryDisplay = null;
     if (!badEffectiveRunCoverage) {
         if (explicitlyNotBadEffectiveRunCoverage === 1) {
-            qcSummaryDisplay = getQcSummaryDisplayWithColor(notBadPercentageFormat, { color: QC_SUMMARY_COLORS.ALL_GOOD });
+            qcSummaryDisplay = getQcSummaryDisplayWithColor(qcSummaryContent, { color: QC_SUMMARY_COLORS.ALL_GOOD });
         } else {
-            qcSummaryDisplay = getQcSummaryDisplayWithColor(notBadPercentageFormat, { color: QC_SUMMARY_COLORS.PARTIALLY_GOOD_NOT_BAD });
+            qcSummaryDisplay = getQcSummaryDisplayWithColor(qcSummaryContent, { color: QC_SUMMARY_COLORS.PARTIALLY_GOOD_NOT_BAD });
         }
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/views/Runs/ActiveColumns/runDetectorsAsyncQcActiveColumns.js]: refactored

File: lib/public/views/Runs/ActiveColumns/runDetectorsAsyncQcActiveColumns.js [hunk 4]
```diff
@@ -67,15 +72,26 @@ const runDetectorAsyncQualityDisplay = ({ dataPassId, simulationPassId }, run, d
     let qcSummaryDisplay = null;
     if (!badEffectiveRunCoverage) {
         if (explicitlyNotBadEffectiveRunCoverage === 1) {
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/views/Runs/ActiveColumns/runDetectorsAsyncQcActiveColumns.js]: applied

File: lib/public/views/Runs/ActiveColumns/runDetectorsAsyncQcActiveColumns.js [hunk 5]
```diff
@@ -67,15 +72,26 @@ const runDetectorAsyncQualityDisplay = ({ dataPassId, simulationPassId }, run, d
...
-            qcSummaryDisplay = getQcSummaryDisplayWithColor(notBadPercentageFormat, { color: QC_SUMMARY_COLORS.ALL_GOOD });
+            qcSummaryDisplay = getQcSummaryDisplayWithColor(qcSummaryContent, { color: QC_SUMMARY_COLORS.ALL_GOOD });
         } else {
-            qcSummaryDisplay = getQcSummaryDisplayWithColor(notBadPercentageFormat, { color: QC_SUMMARY_COLORS.PARTIALLY_GOOD_NOT_BAD });
+            qcSummaryDisplay = getQcSummaryDisplayWithColor(qcSummaryContent, { color: QC_SUMMARY_COLORS.PARTIALLY_GOOD_NOT_BAD });
         }
     } else {
-        if (badEffectiveRunCoverage === 1) {
-            qcSummaryDisplay = getQcSummaryDisplayWithColor(notBadPercentageFormat, { color: QC_SUMMARY_COLORS.ALL_BAD });
+        if (!explicitlyNotBadEffectiveRunCoverage && mcReproducible) {
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/public/views/Runs/ActiveColumns/runDetectorsAsyncQcActiveColumns.js]: fixed

File: lib/server/services/qualityControlFlag/QcFlagService.js [hunk 6]
```diff
@@ -66,21 +66,28 @@ const validateUserDetectorAccess = (userRoles, detectorName) => {
...
     explicitlyNotBadEffectiveRunCoverage: 'explicitlyNotBadEffectiveRunCoverage',
     missingVerificationsCount: 'missingVerificationsCount',
+    mcReproducible: 'mcReproducible',
 };
 
+/**
+ * Id of flag type in DB
+ */
+const LIMITED_ACCEPTANCE_MC_REPORODUCIBLE_FLAG_TYPE_ID = 5;
+
```
[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/server/services/qualityControlFlag/QcFlagService.js:82]: name doesn't make much sense than id, it's also mixing with DB, beside name is alterable via QC flag types API

[the remark this answers is not in the document]
Comment (xsalonx) [reply lib/server/services/qualityControlFlag/QcFlagService.js]: added new column

File: lib/public/views/Runs/ActiveColumns/runDetectorsAsyncQcActiveColumns.js [hunk 7]
```diff
@@ -66,17 +74,32 @@ const runDetectorAsyncQualityDisplay = ({ dataPassId, simulationPassId }, run, d
...
+                color: explicitlyNotBadEffectiveRunCoverage === FULL_COVERAGE
+                    ? QC_SUMMARY_COLORS.ALL_GOOD
+                    : QC_SUMMARY_COLORS.PARTIALLY_GOOD_NOT_BAD,
+            },
+        );
+    } else if (!explicitlyNotBadEffectiveRunCoverage && mcReproducible) {
+        qcSummaryDisplay = getQcSummaryDisplayWithColor(
+            [
+                notBadPercentageFormat,
+                h('.d-inline-block.va-top.f7.italic', 'MC.R'),
```
Comment (martinboulais) [inline lib/public/views/Runs/ActiveColumns/runDetectorsAsyncQcActiveColumns.js:89]: why not using `em` if what you need is emphasis?

File: lib/server/services/qualityControlFlag/QcFlagService.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (martinboulais) [reply lib/server/services/qualityControlFlag/QcFlagService.js]: That is way better

File: lib/public/views/Runs/ActiveColumns/runDetectorsAsyncQcActiveColumns.js (hunk 7 above)
Comment (xsalonx) [reply lib/public/views/Runs/ActiveColumns/runDetectorsAsyncQcActiveColumns.js]: changed

**Review by martinboulais — APPROVED** [review]

**Pull request merged**